extensions/wikia/CreateNewWiki/tasks/CreateDatabase.php

```diff
@@ -47,7 +47,12 @@ public function run() {
 		$dbw = wfGetDB( DB_MASTER, [ ], $this->clusterDB );
 		$this->taskContext->setWikiDBW($dbw);
 
-		$dbw->query( sprintf( "CREATE DATABASE `%s`", $this->taskContext->getDBname() ) );
+		$safeDbName = $dbw->addIdentifierQuotes( $this->taskContext->getDbName() );
+
+		$dbw->query( "CREATE DATABASE $safeDbName" );
+		$dbw->commit( __METHOD__ );
+
+		$this->waitUntilDatabaseExistsOnSlave();
```

Comment:
We need to make sure that the database exists on the cluster slave we're connected to when we move on to the next steps (creating tables). wfWaitForSlaves is expensive (can take over 30+ seconds), so let's just ensure the status of the slave we are connected to, as subsequent wfGetDB invocation for this database during this request will return that connection.